### PR TITLE
🌐 Add French translation for Alternatives, Inspiration and Comparisons

### PR DESCRIPTION
**Pull request by rjNemo**
Hello :wave,

this PR adds 🇫🇷  French translation for `docs/alternatives.md` (issue #1972).

Thank you!

### CODE REVIEW

File: docs/fr/docs/alternatives.md [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+    Il a clairement inspiré Uvicorn et Starlette, qui sont actuellement plus rapides que Sanic dans les benchmarks.
+
+!!! check "A inspiré **FastAPI** à"
+    Trouvez un moyen d'avoir une performance folle.
+    
+    C'est pourquoi **FastAPI** est basé sur Starlette, car il s'agit du framework le plus rapide disponible (testé par des benchmarks tiers).
+
+### <a href="https://falconframework.org/" class="external-link" target="_blank">Falcon</a>
+
+Falcon est un autre framework Python haute performance, il est conçu pour être minimal, et fonctionner comme fondation d'autres frameworks comme Hug.
```
Comment (Jefidev) [inline docs/fr/docs/alternatives.md:266]:
```suggestion
Falcon est un autre framework Python haute performance, il est conçu pour être minimal, et est utilisé comme fondation pour d'autres frameworks comme Hug.
```

File: docs/fr/docs/alternatives.md [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+Il utilise le standard précédent pour les frameworks web Python (WSGI) qui est synchrone, donc il ne peut pas gérer les WebSockets et d'autres cas d'utilisation. Néanmoins, il offre de très bonnes performances.
+
+Il est conçu pour avoir des fonctions qui reçoivent deux paramètres, une "requête" et une "réponse". Ensuite, vous 
+"lisez" des parties de la requête et "écrivez" des parties dans la réponse. En raison de cette conception, il n'est 
+pas possible de déclarer des paramètres de requête et des corps avec des indications de type Python standard comme paramètres de fonction.
+
+Ainsi, la validation, la sérialisation et la documentation des données doivent être effectuées dans le code, et non pas automatiquement. Ou bien elles doivent être implémentées comme un framework au-dessus de Falcon, comme Hug. Cette même distinction se retrouve dans d'autres frameworks qui s'inspirent de la conception de Falcon, qui consiste à avoir un objet de requête et un objet de réponse comme paramètres.
+
+!!! check "A inspiré **FastAPI** à"
+    Trouvez des moyens d'obtenir de bonnes performances.
```
Comment (Jefidev) [inline docs/fr/docs/alternatives.md:277]:
```suggestion
    Trouver des moyens d'obtenir de bonnes performances.
```

File: docs/fr/docs/alternatives.md [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+    Avec Hug (puisque Hug est basé sur Falcon), **FastAPI** a inspiré la déclaration d'un paramètre `response` dans les fonctions.
+
+    Bien que dans FastAPI, il est facultatif, et est utilisé principalement pour définir les en-têtes, les cookies, et les codes de statut alternatifs.
+
+### <a href="https://moltenframework.com/" class="external-link" target="_blank">Molten</a>
+
+J'ai découvert Molten lors des premières étapes de développement de **FastAPI**. Et il a des idées assez similaires :
+
+* Basé sur les type hints Python.
+* Validation et documentation de ces types.
```
Comment (Jefidev) [inline docs/fr/docs/alternatives.md:288]:
```suggestion
* Validation et documentation via ces types.
``` 
*via* may be replaced by *basé sur*

File: docs/fr/docs/alternatives.md [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+* Système d'injection de dépendances.
+
+Il n'utilise pas une package tiers de validation, sérialisation et de documentation tel que Pydantic, il a la sienne. Ainsi, ces définitions de types de données ne seraient pas réutilisables aussi facilement.
```
Comment (Jefidev) [inline docs/fr/docs/alternatives.md:291]:
```suggestion
Il n'utilise pas une librairie tiers de validation, sérialisation et de documentation tel que Pydantic, il utilise son propre système. Ainsi, ces définitions de types de données ne sont pas réutilisables aussi facilement.
```

File: docs/fr/docs/alternatives.md [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+
+Il nécessite une configuration un peu plus verbeuses. Et comme il est basé sur WSGI (au lieu dASGI), il n'est pas
```
Comment (Jefidev) [inline docs/fr/docs/alternatives.md:293]:
```suggestion
Il nécessite une configuration un peu plus verbeuses. Et comme il est basé sur WSGI (au lieu d'ASGI), il n'est pas 
conçu pour profiter des hautes performances fournies par des outils comme Uvicorn, Starlette et Sanic.
```

File: docs/fr/docs/alternatives.md [hunk 6]
```diff
@@ -0,0 +1,454 @@
...
+
+Il n'utilise pas une package tiers de validation, sérialisation et de documentation tel que Pydantic, il a la sienne. Ainsi, ces définitions de types de données ne seraient pas réutilisables aussi facilement.
+
+Il nécessite une configuration un peu plus verbeuses. Et comme il est basé sur WSGI (au lieu dASGI), il n'est pas 
+conçu pour profiter des hautes performances fournies par des outils comme Uvicorn, Starlette et Sanic.
+
+Le système d'injection de dépendances exige le pré-enregistrement des dépendances et les dépendances sont résolues sur la base des types déclarés. Ainsi, il n'est pas possible de déclarer plus d'un "composant" qui fournit un certain type.
+
+Les routes sont déclarées à un seul endroit, en utilisant des fonctions déclarées à d'autres endroits (au lieu 
+d'utiliser des décorateurs qui peuvent être placés juste au-dessus de la fonction qui gère lendpoint). Cette
```
Comment (Jefidev) [inline docs/fr/docs/alternatives.md:299]:
```suggestion
d'utiliser des décorateurs qui peuvent être placés juste au-dessus de la fonction qui gère l'endpoint). Cette 
```

File: docs/fr/docs/alternatives.md [hunk 7]
```diff
@@ -0,0 +1,454 @@
...
+
+    Je considère **FastAPI** comme un "successeur spirituel" d'APIStar, tout en améliorant et en augmentant les 
+fonctionnalités, le système de typage et d'autres parties, sur la base des enseignements tirés de tous ces outils 
+précédents.
+
+## Utilisé par **FastAPI**
+
+### <a href="https://pydantic-docs.helpmanual.io/" class="external-link" target="_blank">Pydantic</a>
+
+Pydantic est une bibliothèque permettant de définir la validation, la sérialisation et la documentation des données (à l'aide de JSON Schema) en se basant sur les indices de type Python.
```
Comment (Jefidev) [inline docs/fr/docs/alternatives.md:384]:
```suggestion
Pydantic est une bibliothèque permettant de définir la validation, la sérialisation et la documentation des données (à l'aide de JSON Schema) en se basant sur les Python type hints.
``` 
To stay consistent with the rest of the translation were you chose to keep the english version. We could brainstorm on a better french translation for 'Python type hints' if you think it is needed

File: docs/fr/docs/alternatives.md [hunk 8]
```diff
@@ -0,0 +1,454 @@
...
+basé sur les mêmes type hints Python, le support de l'éditeur est grand.
+
+!!! check "**FastAPI** l'utilise pour"
+    Gérer toute la validation des données, leur sérialisation et la documentation automatique du modèle (basée sur le schéma JSON).
+
+    **FastAPI** prend ensuite ces données JSON Schema et les place dans OpenAPI, en plus de toutes les autres choses qu'il fait.
+
+### <a href="https://www.starlette.io/" class="external-link" target="_blank">Starlette</a>
+
+Starlette est un léger framework/toolkit <abbr title="The new standard for building asynchronous Python web">ASGI</abbr>, qui est idéal pour construire des services asyncio performants.
```
Comment (Jefidev) [inline docs/fr/docs/alternatives.md:398]:
```suggestion
Starlette est un framework/toolkit léger <abbr title="Le nouveau standard pour construire des applications web assynchrones en Python">ASGI</abbr>, qui est idéal pour construire des services asyncio performants.
```

File: docs/fr/docs/alternatives.md [hunk 9]
```diff
@@ -0,0 +1,454 @@
...
+
+Il est très simple et intuitif. Il est conçu pour être facilement extensible et avoir des composants modulaires.
+
+Il offre :
+
+* Des performances vraiment impressionnantes.
+* Le support des WebSockets.
+* Le support de GraphQL.
+* Les tâches d'arrière-plan en cours de traitement.
```
Comment (Jefidev) [inline docs/fr/docs/alternatives.md:407]:
```suggestion
* Les tâches d'arrière-plan
```
In my translation of features.md I had the same sentence to translate and I just said "Tâche d'arrière plan" as I had no idea how to translate "background task" in this context.

File: docs/fr/docs/alternatives.md [hunk 10]
```diff
@@ -0,0 +1,454 @@
...
+* Les événements de démarrage et d'arrêt.
+* Un client de test basé sur request.
```
Comment (Jefidev) [inline docs/fr/docs/alternatives.md:409]: In my translation of features.md I chose to translate it like this : "Test client construits sur les requêtes"

**Review by Jefidev — CHANGES_REQUESTED** [review]
Thanks for the translation. I pinpointed several typos and small mistakes.
I think it could be a good idea to read again the document starting by the end as I was tired when I read it and I may have missed some mistakes.

File: docs/fr/docs/alternatives.md (hunk 10 above)
Comment (rjNemo) [reply docs/fr/docs/alternatives.md:409]: I understand but here request stands for the library named request

File: docs/fr/docs/alternatives.md [hunk 11]
```diff
@@ -0,0 +1,454 @@
...
+marshalling, conversion">sérialisation</abbr>" des données, qui consiste à prendre les données du code (Python) et à 
+les convertir en quelque chose qui peut être envoyé sur le réseau. Par exemple, convertir un objet contenant des 
+données provenant d'une base de données en un objet JSON. Convertir des objets `datetime` en strings, etc.
+
+La validation des données est une autre fonctionnalité importante dont ont besoin les API. Elle permet de s'assurer 
+que les données sont valides, compte tenu de certains paramètres. Par exemple, qu'un champ est un `int`, et non un 
+string.
+Ceci est particulièrement utile pour les données entrantes.
+
+Sans un système de validation des données, vous devriez effectuer toutes les vérifications à la main, en code.
```
Comment (Smlep) [inline docs/fr/docs/alternatives.md:151]:
```suggestion
Sans un système de validation des données, vous devriez effectuer toutes les vérifications à la main, dans le code.
```

File: docs/fr/docs/alternatives.md [hunk 12]
```diff
@@ -0,0 +1,454 @@
...
+
+Ces fonctionnalités sont ce pourquoi Marshmallow a été construit. C'est une excellente librairie, et je l'ai déjà beaucoup utilisée.
```
Comment (Smlep) [inline docs/fr/docs/alternatives.md:153]:
```suggestion
Ces fonctionnalités sont ce pourquoi Marshmallow a été construit. C'est une excellente bibliothèque, et je l'ai déjà beaucoup utilisée.
```

File: docs/fr/docs/alternatives.md [hunk 13]
```diff
@@ -0,0 +1,454 @@
...
+Maintenant, APIStar est un ensemble d'outils pour valider les spécifications OpenAPI, et non un framework web.
+
+!!! info
+    APIStar a été créé par Tom Christie. Le même gars qui a créé :
+
+    * Django REST Framework
+    * Starlette (sur lequel **FastAPI** est basé)
+    * Uvicorn (utilisé par Starlette et **FastAPI**)
+
+!!! check "Inspired **FastAPI** to"
```
Comment (Smlep) [inline docs/fr/docs/alternatives.md:367]:
```suggestion
!!! check "A inspiré **FastAPI** à"
```

File: docs/fr/docs/alternatives.md [hunk 14]
```diff
@@ -0,0 +1,454 @@
...
+    Exister.
+
+    L'idée de déclarer plusieurs choses (validation des données, sérialisation et documentation) avec les mêmes types Python, tout en offrant un excellent support pour les éditeurs, était pour moi une idée brillante.
+    
+    Et après avoir longtemps cherché un framework similaire et testé de nombreuses alternatives, APIStar était la meilleure option disponible.
+
+    Puis APIStar a cessé d'exister en tant que serveur et Starlette a été créé, et a constitué une meilleure base pour un tel système. Ce fut l'inspiration finale pour construire **FastAPI**.
+
+    Je considère **FastAPI** comme un "successeur spirituel" d'APIStar, tout en améliorant et en augmentant les 
+fonctionnalités, le système de typage et d'autres parties, sur la base des enseignements tirés de tous ces outils
```
Comment (Smlep) [inline docs/fr/docs/alternatives.md:377]: The linebreak with no indenting breaks the formatting
image

File: docs/fr/docs/alternatives.md [hunk 15]
```diff
@@ -0,0 +1,454 @@
...
+    
+    Et après avoir longtemps cherché un framework similaire et testé de nombreuses alternatives, APIStar était la meilleure option disponible.
+
+    Puis APIStar a cessé d'exister en tant que serveur et Starlette a été créé, et a constitué une meilleure base pour un tel système. Ce fut l'inspiration finale pour construire **FastAPI**.
+
+    Je considère **FastAPI** comme un "successeur spirituel" d'APIStar, tout en améliorant et en augmentant les 
+fonctionnalités, le système de typage et d'autres parties, sur la base des enseignements tirés de tous ces outils 
+précédents.
+
+## Utilisé par **FastAPI**
```
Comment (Smlep) [inline docs/fr/docs/alternatives.md:380]:
```suggestion
## Utilisés par **FastAPI**
```

File: docs/fr/docs/alternatives.md [hunk 16]
```diff
@@ -0,0 +1,454 @@
...
+Starlette est un framework/toolkit léger <abbr title="Le nouveau standard pour construire des applications web assynchrones en Python">ASGI</abbr>, qui est idéal pour construire des services asyncio performants.
+
+Il est très simple et intuitif. Il est conçu pour être facilement extensible et avoir des composants modulaires.
+
+Il offre :
+
+* Des performances vraiment impressionnantes.
+* Le support des WebSockets.
+* Le support de GraphQL.
+* Les tâches d'arrière-plan
```
Comment (Smlep) [inline docs/fr/docs/alternatives.md:407]:
```suggestion
* Les tâches d'arrière-plan.
```

**Review by Smlep — APPROVED** [review]

File: docs/fr/docs/alternatives.md [hunk 17]
```diff
@@ -0,0 +1,453 @@
...
+La validation des données est une autre fonctionnalité importante dont ont besoin les API. Elle permet de s'assurer 
+que les données sont valides, compte tenu de certains paramètres. Par exemple, qu'un champ est un `int`, et non un 
+string.
+Ceci est particulièrement utile pour les données entrantes.
+
+Sans un système de validation des données, vous devriez effectuer toutes les vérifications à la main, dans le code.
+
+Ces fonctionnalités sont ce pourquoi Marshmallow a été construit. C'est une excellente bibliothèque, et je l'ai déjà beaucoup utilisée.
+
+Mais elle a été créé avant que les type hints n'existent en Python. Ainsi, pour définir chaque <abbr title="la définition de
```
Comment (NastasiaSaby) [inline docs/fr/docs/alternatives.md:155]: It should be "créée".

File: docs/fr/docs/alternatives.md [hunk 18]
```diff
@@ -0,0 +1,453 @@
...
+la façon dont les données doivent être formées">schéma</abbr>, vous devez utiliser des utilitaires et des classes spécifiques fournis par Marshmallow.
```
Comment (NastasiaSaby) [inline docs/fr/docs/alternatives.md:156]: It should be "classes spécifiques fournies"

File: docs/fr/docs/alternatives.md [hunk 19]
```diff
@@ -0,0 +1,453 @@
...
+
+J'ai découvert Molten lors des premières étapes de développement de **FastAPI**. Et il a des idées assez similaires :
+
+* Basé sur les type hints Python.
+* Validation et documentation via ces types.
+* Système d'injection de dépendances.
+
+Il n'utilise pas une librairie tiers de validation, sérialisation et de documentation tel que Pydantic, il utilise son propre système. Ainsi, ces définitions de types de données ne sont pas réutilisables aussi facilement.
+
+Il nécessite une configuration un peu plus verbeuses. Et comme il est basé sur WSGI (au lieu d'ASGI), il n'est pas
```
Comment (NastasiaSaby) [inline docs/fr/docs/alternatives.md:293]: "verbeuse" instead of "verbeuses"

File: docs/fr/docs/alternatives.md [hunk 20]
```diff
@@ -0,0 +1,453 @@
...
+
+* Basé sur les type hints Python.
+* Validation et documentation via ces types.
+* Système d'injection de dépendances.
+
+Il n'utilise pas une librairie tiers de validation, sérialisation et de documentation tel que Pydantic, il utilise son propre système. Ainsi, ces définitions de types de données ne sont pas réutilisables aussi facilement.
+
+Il nécessite une configuration un peu plus verbeuses. Et comme il est basé sur WSGI (au lieu d'ASGI), il n'est pas 
+conçu pour profiter des hautes performances fournies par des outils comme Uvicorn, Starlette et Sanic.
+conçu pour profiter des hautes performances fournies par des outils comme Uvicorn, Starlette et Sanic.
```
Comment (NastasiaSaby) [inline docs/fr/docs/alternatives.md:295]: This line is duplicated, isn't it?

File: docs/fr/docs/alternatives.md [hunk 21]
```diff
@@ -0,0 +1,453 @@
...
+
+Comme il est basé sur l'ancienne norme pour les frameworks web Python synchrones (WSGI), il ne peut pas gérer les Websockets et autres, bien qu'il soit également très performant.
+
+!!! info
+    Hug a été créé par Timothy Crosley, le créateur de <a href="https://github.com/timothycrosley/isort" class="external-link" target="_blank">`isort`</a>, un excellent outil pour trier automatiquement les imports dans les fichiers Python.
+
+!!! check "A inspiré **FastAPI** à"
+    Hug a inspiré certaines parties d'APIStar, et était l'un des outils que je trouvais les plus prometteurs, à côté d'APIStar.
+
+    Hug a contribué à inspirer **FastAPI** pour utiliser les type hints Python
```
Comment (NastasiaSaby) [inline docs/fr/docs/alternatives.md:330]: Maybe a "." at the end of the sentence.

File: docs/fr/docs/alternatives.md [hunk 22]
```diff
@@ -0,0 +1,453 @@
...
+* Le support des WebSockets.
+* Le support de GraphQL.
+* Les tâches d'arrière-plan.
+* Les événements de démarrage et d'arrêt.
+* Un client de test basé sur request.
+* CORS, GZip, fichiers statiques, streaming des réponses.
+* Le support des sessions et des cookies.
+* Une couverture de test à 100 %.
+* 100 % de la base de code avec des annotations de type.
+* Zéro fortes dépendances à d'autres packages.
```
Comment (NastasiaSaby) [inline docs/fr/docs/alternatives.md:413]: Maybe "zéro forte dépendance".

File: docs/fr/docs/alternatives.md (hunk 21 above)
Comment (rjNemo) [reply docs/fr/docs/alternatives.md:323]: The sentence does not end here

**Review by NastasiaSaby — APPROVED** [review]

**Comment by tiangolo** [comment]
Thanks for the contribution @rjNemo ! :rocket: 

And thanks for the reviews and help @Jefidev, @NastasiaSaby, and @Smlep ! :bow: :coffee:

**Pull request merged**